src/rngs/adapter/reseeding.rs

```diff
@@ -181,9 +189,15 @@ where R: BlockRngCore + SeedableRng,
     fn reseed_and_generate(&mut self,
                            results: &mut <Self as BlockRngCore>::Results)
     {
-        trace!("Reseeding RNG after {} generated bytes",
-               self.threshold - self.bytes_until_reseed);
-        let threshold = if let Err(e) = self.reseed()  {
+        let fork_counter = get_fork_counter();
+        if self.fork_counter < fork_counter {
+            warn!("Fork detected, reseeding RNG");
```

Comment:
Probably `info` or `trace` is more appropriate. It's not actually a "problem". In a way it would be nice to use the same level for both.